highway.tex

```diff
@@ -24,807 +26,1323 @@
+% More formally, we recursively define the function $\vote_k$, corresponding to the vote on $k$-th block in the chain, as follows:
 
-\begin{lemma}\label{SummitLemma}
-Assume $p$ is $\ww$-majority-driven relative to $p'$. Let $(C_0, C_1, \ldots)$ be a summit, $l > 0$, $\mu$ a $<$-minimal $p'$-message with $\neg p(\mu)$ and $J(\mu) \cap C_k \neq \emptyset$. Let
-$$B_k(\mu) = \{S(\zeta) \mid \text{ $\zeta < \mu$, $p'(\zeta)$, $\neg p(\zeta)$, and there exists $C'_{k-1} \ni \nu' \sqsubset \zeta$ but no $C_k \ni \nu \sqsubset \zeta$.}\}$$
-be the set of validators that voted against $p$ after sending a $C'_{k-1}$ message. Then:
-$$\ww(B_k(\mu) \setminus E(\mu)) + \frac{1}{2} \ww(E(\mu)) \;\geq\; t$$
-\end{lemma}
+% \begin{itemize}
+%     \item $\vote_0(u) = G$
+%     \item $\vote_k(u) = \min \left(\argmax\limits_{B\in\mathrm{next}(\vote_{k-1}(u))} \ww(\{v \in L (\bar{D}(u)) \mid \vote_{k}(v) = B\})\right)\textrm{,}$
+% \end{itemize}
 
-The intuition behind this is that a $\lambda \in J(\mu) \cap C_k$ saw a majority of $q = \ww(\mathcal{V})/2 + t$ for $p$, but $\mu$ does not see such a majority anymore.
-This can happen because $t$ of those validators (by weight) changed their vote away from $p$.
-It can also happen because $s$ validators equivocate:
-Those don't count as voting for $p$ anymore, but they also don't count as voters at all, so anything greater than $(\ww(\mathcal{V}) - s)/2$ would still be a majority.
-That's why only $s = 2t$ equivocators could bring down the votes for $p$ below the new majority threshold $\ww(\mathcal{V}) / 2 - t$.
-Of course the reason can also be a combination of equivocations and changed votes.
 
-\begin{proof}
-Since $\neg p(\mu)$, $p$ cannot have the majority in $L_{p'}(\mu)$, i.e.:
-\begin{flalign*}
-\ww(\{ \nu \in L_{p'} (\mu) \mid p(\nu) \}) &\leq \frac{1}{2} \ww(L_{p'}(\mu)) \leq \frac{1}{2} \ww(\mathcal{V}) - \frac{1}{2} \ww(E(\mu))
-\end{flalign*}
-On the other hand, if $\lambda \in J(\mu) \cap C_k$, $\ww(\bar{J}(\lambda) \cap C'_{k-1}) \geq \frac{1}{2} \ww(\mathcal{V}) + t$, and every validator in $S(\bar{J}(\lambda) \cap C'_{k-1})$ who does not count as a vote for $p$ in $\mu$ must either have equivocated or produced a vote $\zeta$ against $p$ after their message in $C'_{k-1}$. In the latter case, if $\zeta$ could see $S(\zeta)$'s $C_k$ message, $\mu$ wouldn't be minimal; thus such an $S(\zeta)$ would be in $B_k(\mu)$:
-\begin{flalign*}
-\ww(\{ \nu \in L_{p'} (\mu) \mid p(\nu) \}) & \geq \ww(S(\bar{J}(\lambda) \cap C'_{k-1}) \setminus E(\mu) \setminus B_k(\mu)) &\\
-& \geq \ww(\bar{J}(\lambda) \cap C'_{k-1}) - \ww(E(\mu) \cup B_k(\mu)) &\\
-& \geq \frac{1}{2} \ww(\mathcal{V}) + t - \ww(E(\mu)) -\ww(B_k(\mu) \setminus E(\mu)))&
-\end{flalign*}
-Together, these prove the inequality.
-\end{proof}
+\subsection{Finality condition}\label{sec:finality}
+% \damian{Would be good to write some prose here.}
```

Comment:
Let's not merge TODO comments, except if the plan is to make a follow-up PR soon.